src/StreamJsonRpc.Shared/JsonRpc.cs

```diff
@@ -655,6 +667,69 @@ protected virtual Task<ReturnType> InvokeCoreAsync<ReturnType>(int? id, string t
             }
         }
 
+        /// <summary>
+        /// Creates a dictionary which maps a request method name to its clr method name via <see cref="JsonRpcMethodAttribute" /> value.
+        /// </summary>
+        /// <param name="target">Object to reflect over and analyze its methods.</param>
+        /// <returns>Dictionary which maps a request method name to its clr method name.</returns>
+        private void GetRequestMethodToClrMethodMap(object target)
+        {
+            if (target != null)
+            {
+                for (TypeInfo t = target.GetType().GetTypeInfo(); t != null; t = t.BaseType?.GetTypeInfo())
+                {
+                    foreach (MethodInfo method in t.DeclaredMethods)
+                    {
+                        var attribute = (JsonRpcMethodAttribute)method.GetCustomAttribute(typeof(JsonRpcMethodAttribute));
+                        var attributeValue = attribute == null ? null : attribute.Name;
+
+                        string requestMethodValue;
+                        if (clrMethodToRequestMethodMap.TryGetValue(method.Name, out requestMethodValue))
+                        {
+                            if (!string.Equals(requestMethodValue, attributeValue, StringComparison.Ordinal))
+                            {
+                                Requires.Fail(Resources.ConflictingMethodNameAttribute, method.Name, nameof(JsonRpcMethodAttribute), nameof(JsonRpcMethodAttribute.Name));
+                            }
+                        }
+                        else
+                        {
+                            clrMethodToRequestMethodMap.Add(method.Name, attributeValue);
+                        }
+
+                        if (attributeValue != null)
```

Comment:
Why not unconditionally add the requestMethodToClrMethodMap entry? We always need to do that mapping at runtime for incoming requests, and it seems to me like if we guaranteed an entry in this map, we wouldn't need to keep the second dictionary around.